### PR TITLE
[Ft: #155326153] Add Reset password mutation

### PR DESCRIPTION
**Pull request by fisayomi**
#### What does this PR do?
- Adds a separate mutation to reset password if forgotten 

#### How should this be manually tested?
- run the added resetPassword mutation

#### What are the relevant pivotal tracker stories?
[#155326153](https://www.pivotaltracker.com/story/show/155326153)

### CODE REVIEW

**Comment by omorilewa** [comment]
Good work ma'am. Could we also confirm it works well in staging.

**Comment by omorilewa** [comment]
Also, noticed a story ugonna picked is similar to this. Is it still necessary for her to go ahead with hers?

**Comment by oreoluwade** [comment]
👍

File: food_basket/schema/user.py [hunk 1]
```diff
@@ -123,6 +123,35 @@ def mutate(root, info, email):
...
+        try:
+            user = User.objects.get(id=user_id)
+            if getattr(user, 'is_active'):
+                token_object = Token.objects.get(key=args.get('token'))
+                if user_id != getattr(token_object, 'user_id'):
+                    return ResetPassword(message="Unauthorized user")
+
+                user.set_password(password)
+                user.save()
+                return UpdateUser(message="Password reset Successful!")
```
Comment (omorilewa) [inline food_basket/schema/user.py]: Should this be `UpdateUser`?

**Comment by oreoluwade** [comment]
@omorilewa That story is about adding the password field to the `UpdateUser` mutation (In the case that someone wants to change their password even without forgetting it). It could still be continued.

File: food_basket/schema/user.py (hunk 1 above)
Comment (oreoluwade) [reply food_basket/schema/user.py]: 😆

**Comment by omorilewa** [comment]
@oreoluwade Thanks. I get the difference now.

**Comment by fisayomi** [comment]
@omorilewa How can we confirm it works well in staging?

**Comment by oreoluwade** [comment]
@fisayomi Should be by deploying that branch and testing the mutation before merging it (I think).

**Comment by omorilewa** [comment]
@fisayomi, @oreoluwade said what I had in mind. Perhaps we should start to look at having review apps on heroku such that we are able to test the PR changes before merging.

**Comment by fisayomi** [comment]
It's working well in staging @omorilewa @oreoluwade .

**Pull request merged**